client/trino-jdbc/src/main/java/io/trino/jdbc/ConnectionProperties.java

```diff
@@ -513,6 +516,15 @@ public KerberosDelegation()
         }
     }
 
+    private static class KerberosConstrainedDelegation
+            extends AbstractConnectionProperty<GSSCredential, GSSCredential>
+    {
+        public KerberosConstrainedDelegation()
+        {
+            super("KerberosConstrainedDelegation", Optional.empty(), NOT_REQUIRED, ALLOWED, GSSCredential.class::cast);
```

Comment:
Should we apply the check like `isKerberosEnabled` and should we add any additional validation ?

Reply:
Or `isKerberosWithoutDelegation`